### ISSUE TITLE
Clean up leaky interface that requires `discrete_time_problem` to know about time freezing.

### ISSUE THREAD

**Issue by apozharski** [nurkanovic/nosnoc#117]
Currently `Stewart` and `Heaviside` problems use `Options.time_freezing` to identify a physical time state.  This makes the resulting implementation confusing. A possible solution to this is just adding a new option that specifies the existence of a physical time state.

**Comment by nurkanovic** [comment]
I agree. And the default is of course that the clock state does not exist.
